dandi/validate.py

```diff
@@ -44,12 +92,89 @@ def validate_bids(
         report_path = report_path.format(
             log_dir=log_dir,
         )
+
     validation_result = validate_bids_(
         paths,
         schema_version=schema_version,
         report_path=report_path,
     )
-    return dict(validation_result)
+    our_validation_result = []
+    origin = ValidationOrigin(
+        name="bidsschematools",
+        version=bidsschematools.__version__,
+    )
+
+    # Storing variable to not re-compute set paths for each individual file.
+    parent_path = None
+    for path in validation_result["path_tracking"]:
+        # Hard-coding exclusion here pending feature + release in:
+        # https://github.com/bids-standard/bids-specification/issues/1272
+        if path.endswith((".ERRORS", ".ERRORS.json")):
+            continue
+        if parent_path != os.path.dirname(path):
+            parent_path = os.path.dirname(path)
+            dataset_path = find_parent_directory_containing("dataset_description.json", parent_path)
+            dandiset_path = find_parent_directory_containing("dandiset.yaml", parent_path)
+        our_validation_result.append(
+            ValidationResult(
+                origin=origin,
+                severity=Severity.ERROR,
+                id="BIDS.NON_BIDS_PATH_PLACEHOLDER",
+                scope=Scope.FILE,
+                path=Path(path),
+                message="File does not match any pattern known to BIDS.",
+                dataset_path=dataset_path,
+                dandiset_path=dandiset_path,
+            )
+        )
+
+    for pattern in validation_result["schema_tracking"]:
+        # Future proofing for standard-compliant name.
+        if pattern.get("mandatory") or pattern.get("required"):
+            # We don't have a path for this so we'll need some external logic to make sure
+            # that the dataset path is populated.
+            # dataset_path = find_parent_directory_containing(paths, path)
+            our_validation_result.append(
+                ValidationResult(
+                    origin=origin,
+                    severity=Severity.ERROR,
+                    id="BIDS.MANDATORY_FILE_MISSING_PLACEHOLDER",
+                    scope=Scope.DATASET,
+                    path_regex=pattern["regex"],
+                    message="BIDS-required file is not present.",
+                )
+            )
+
+    # Storing variable to not re-compute set paths for each individual file.
+    parent_path = None
+    for meta in validation_result["match_listing"]:
+        file_path = meta.pop("path")
+        meta = {BIDS_TO_DANDI[k]: v for k, v in meta.items() if k in BIDS_TO_DANDI}
+        if parent_path != os.path.dirname(file_path):
+            parent_path = os.path.dirname(file_path)
+            dataset_path = find_parent_directory_containing("dataset_description.json", parent_path)
+            dandiset_path = find_parent_directory_containing("dandiset.yaml", parent_path)
```

Comment:
pay attention to LGTM (although soon it will die). You overload these two variables in the next two lines below.  So which should it be -- should there be `else:` or just these two removed?